### PR TITLE
Implement calculator

### PR DESCRIPTION
**Pull request by smbryar**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
 console.log('Welcome to the calculator!');
 
-console.log('Please enter some input:');
-const response = readline.prompt();
-console.log('You entered: ' + response);
+console.log('Enter the operator:');
+const operator = readline.prompt();
+
+console.log(`How many numbers do you want to ${operator} ?`);
+const c = parseInt(readline.prompt());
```
Comment (smbryar) [inline index.js:10]: In this file you have used a few very short variable names (c, n, calc). These took me a few moments to work out what they represented. What longer variable names could you use that are clearer?

File: calculator.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          case '-':
+            answer = answer - number;
+            break;
+          case '/':
+            answer = answer / number;
+            break;
+          default:
+            console.log("Invalid operator " + operator);
+        }
+        // console.log('index: ' + index);
```
Comment (smbryar) [inline calculator.js:38]: A couple of commented out lines to remove here :)

File: index.js [hunk 3]
```diff
@@ -1,7 +1,20 @@
 const readline = require('readline-sync');
+const calc = require('./calculator');
 
 console.log('Welcome to the calculator!');
 
-console.log('Please enter some input:');
-const response = readline.prompt();
-console.log('You entered: ' + response);
+console.log('Enter the operator:');
```
Comment (smbryar) [inline index.js:6]: I can imagine someone typing 'plus' here and being unsure why the operator is invalid. Can you add the operators that are allowed to the prompt?

File: calculator.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      let answer = numbers[0];
+
+      let i = 0;
+      numbers.forEach(function (number) {
+        if (i === 0) {
+          i++;
+          return;
+        }
+
+        switch (operator) {
```
Comment (smbryar) [inline calculator.js:25]: Looks like you've missed out the '*' operator

File: calculator.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
+
+      let i = 0;
+      numbers.forEach(function (number) {
```
Comment (smbryar) [inline calculator.js:19]: I like that you are using an array method here, but I think that [reduce ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) might be more appropriate as it is designed to work its way through the array element-by-element, combining the current array element with the running total e.g. by adding or multiplying.

```
const finalAnswer= numbers.reduce(
  ( answer, number) => {
    // your great switch function code here
    return answer;
  }
);
```

File: calculator.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+          case '/':
+            answer = answer / number;
+            break;
+          default:
+            console.log("Invalid operator " + operator);
+        }
+        // console.log('index: ' + index);
+        // console.log('current answer: ' + answer);
+
+        i++;
```
Comment (smbryar) [inline calculator.js:41]: If you stuck with using a `forEach` loop, this `i++` is not actually necessary. Could you explain why?

File: test/calculator_test.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+      calculator.add_number(0);
+      calculator.add_number(-20);
+      calculator.add_number(1486);
+      calculator.add_number(-402);
+      calculator.add_number(4);
+
+      assert.equal(1077, calculator.calculate());
+    });
+
+    it('rejects the calculation when there are no numbers', function () {
```
Comment (smbryar) [inline test/calculator_test.js:38]: Great that you are covering this edge case

File: test/calculator_test.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+const assert = require('assert');
+const chai = require('chai');
+const calc = require('../calculator')
+
+describe('calculator', function () {
+  describe('#calculate()', function () {
+    it('adds one number', function () {
```
Comment (smbryar) [inline test/calculator_test.js:7]: I'm unsure if these two cases (adding two numbers and many numbers) are that different from each other, as testing many numbers covers testing two numbers.

What other test cases could be added to cover the cases we expect and potential bad inputs?
e.g. what other operators should we be testing, what happens if the user submits letters instead of numbers by mistake, etc

File: calculator.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
```
Comment (smbryar) [inline calculator.js:1]: Could you use this array of operators to check that the user has submitted a valid operator?
